### PR TITLE
ZIOS-9341 Fix/localized date format

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

In conversation screen, the date string of previous date is not localized. The dates of last year do not have year component.
A static NSDateFormatter would cause year is not shown issue if the first call is for a date in previous year and next call in current year.

### Solutions
Rewrite NSDate+WRFormat in Swift. Created thisYearFormatter and otherYearFormatter to replace elseFormatter. 
Added snapshot tests for timestamp display in en_US, DE and zh_HK localization to show the different date format output.
Fix the snapshots not shot with CET timezone.

### CODE REVIEW

**Comment by mikeger** [comment]
@billypchan please resolve the conflict in order for this PR to get merged.

File: WireExtensionComponents/Utilities/NSDate+WRFormat.m [hunk 1]
```diff
@@ -122,16 +119,7 @@ - (NSString *)wr_formattedDate
...
-            
-            if (isThisYear) {
-                formatString = [NSDateFormatter dateFormatFromTemplate:@"EEEEdMMMM" options:0 locale:locale];
-            }
-            else {
-                formatString = [NSDateFormatter dateFormatFromTemplate:@"EEEEdMMMMYYYY" options:0 locale:locale];
-            }
-            elseFormatter = [[NSDateFormatter alloc] init];
-            [elseFormatter setDateFormat:formatString];
+            elseFormatter = [self localizedDateFormatter];
```
Comment (typfel) [inline WireExtensionComponents/Utilities/NSDate+WRFormat.m]: This `elseFormatter` is created once and saved in a static variable. How is switching between the date format strings (with or without year) supposed to work? Wouldn't the first date we use this method on set the date format string for all future invocations?

Comment (billypchan) [reply WireExtensionComponents/Utilities/NSDate+WRFormat.m]: @mikeger @typfel I will rewrite the elseFormatter. This static NSDateFormatter would change its format during run time. Every time the cell gets NSDate.wr_formattedDate (non static), elseFormatter would changes. 

I would create two static NSDateFormatters to replace elseFormatter, one for date format in the same year, and another one for in previous years case.

File: WireExtensionComponents/Utilities/Date+Format.swift [hunk 2]
```diff
@@ -16,14 +16,128 @@
...
+        let thisWeekComponents: DateComponents? = gregorian.dateComponents(WRDateFormatter.WeekMonthYearUnits, from: today)
+        // Received date
+        let dateComponents: DateComponents? = gregorian.dateComponents(WRDateFormatter.DayMonthYearUnits, from: self)
+        let weekComponents: DateComponents? = gregorian.dateComponents(WRDateFormatter.WeekMonthYearUnits, from: self)
+
+        let intervalSinceDate: TimeInterval = -timeIntervalSinceNow
+        let isToday: Bool = todayDateComponents == dateComponents
+        let isYesterday: Bool = yesterdayComponents == dateComponents
+        let isThisWeek: Bool = thisWeekComponents == weekComponents
+        //        let isThisYear = Calendar.current.isDate(self, equalTo: today, toGranularity: .year)
```
Comment (mikeger) [inline WireExtensionComponents/Utilities/Date+Format.swift]: Commented code.

**Pull request merged**